azure-webapp-maven-plugin/README.md

```diff
@@ -1,186 +1,179 @@
+                                </includes>
+                            </resource>
+                        </resources>
+                    </deployment>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>      
     ```
-   
-3. Use the following commands to deploy your project to Azure App Service. 
-
+3. Use the following commands to deploy your project to Azure App Service.
     ```
     $ mvn azure-webapp:deploy
     ```
 
-<a name="samples"></a>
-## Samples	
-A few typical usages of Maven Plugin for Azure App Service Web Apps are listed at [Web App Samples](../docs/web-app-samples.md). You can choose one to quickly get started.
-
 <a name="goals"></a>
 ## Goals
 
-The Maven Plugin for Azure App Service Web Apps has only one goal: `azure-webapp:deploy`. 
-
-Goal | Description
---- | ---
-`azure-webapp:deploy` | Deploy artifacts or docker container image to an Azure Web App based on your configuration.<br>If the specified Web App does not exist, it will be created.
-## Authentication with Azure
-
-You can use the Azure CLI 2.0 for authentication. More authentication methods can be found [here](../docs/common-configuration.md).  
-
-1. Install the Azure CLI 2.0 by following the instructions in the [Install Azure CLI 2.0](https://docs.microsoft.com/cli/azure/install-azure-cli) article.
-
-2. Run the following commands to log into your Azure subscription:
-
-    ```
-    $ az login
-    $ az account set --subscription <subscription Id>
+#### `azure-webapp:config`
+-  A plugin goal providing a great experience for managing the configuration of [azure-webapp:deploy](#azure-webappdeploy), it will rewrite your pom configuration based on your input. Details about the configuration could be found in the following.
+-  The plugin goal will divide the configuration into 3 parts.
+   1. Application
+        
+        You can configure appName, resource group, region, pricing tier at this part.
+   2. Runtime
+        
+        You can configure operating system and runtime stack at this part. It should be noted that you can not change the operating system after you deploy it.
+   3. DeploymentSlot
+        
+        You can configure deployment slot at this part.
+
+#### `azure-webapp:deploy`
+- Deploy artifacts or docker container image to an Azure Web App based on your configuration.<br>If the specified Web App does not exist, it will be created.
+- Configuration:
+    Common configurations of all Maven Plugins for Azure can be found [here](../docs/common-configuration.md).    
+
+    > This maven plugin supports two kinds of configurations V2 and V1 (deprecated). Using v2 schema is recommended. Please click [here](v1-schema.md) if you want to find more details about v1 schema (deprecated).
+
+    Property | Required | Description | Version
+    ---|---|---|---
+    `<schemaVersion>` | false | Specify the configuration `<schemaVersion>V2</schemaVersion>` to use the V2 configuration. | 1.5.2
```

Comment:
Specify the version of the configuration schema. Supported values are: `v1`, `v2`